mach_eval/analyzers/electromagnetic/bspm/machine_constant/bspm_mach_constants.py

```diff
@@ -0,0 +1,489 @@
+        return iterator
+    
+###########################################################################
+from functools import cached_property, lru_cache
+# @lru_cache decoractor saves the return value of the method, 
+# method will run once and the return value will be saved, from then everytime 
+# the method is called it will simply return the saved return value, 
+# sigificantly reducing computation time
+###########################################################################
+
+from ....force_vector_data import (
+    ProcessForceDataProblem,
+    ProcessForceDataAnalyzer
+)
+
+from ....torque_data import (
+    ProcessTorqueDataProblem,
+    ProcessTorqueDataAnalyzer
+)
+
+class BSPMMachineConstantProblem:
+    def __init__(
+            self,
+            machine:BSPM_Machine,
+            operating_point: BSPM_Machine_Oper_Pt,
+            solve_Kf: bool = True,
+            solve_Kt: bool = True,
+            solve_Kphi: bool = True,
+            solve_Kdelta: bool = True,
+            ) -> 'BSPMMachineConstantProblem':
+        """BSPMMachineConstantProblem Class
+
+        Args:
+            machine (BSPM_Machine): instance of `BSPM_Machine`
+            operating_point (BSPM_Machine_Oper_Pt): instance of `BSPM_Machine_Oper_Pt`
+            solve_Kf (bool, optional): solve force constant. Defaults to True.
+            solve_Kt (bool, optional): solve torque constant. Defaults to True.
+            solve_Kdelta (bool, optional): solve displacment constant. Defaults to True.
+            solve_Kphi (bool, optional): solve back-emf constant. Defaults to True.
+
+        Returns:
+            BSPMMachineConstantProblem: instance of BSPMMachineConstantProblem
+        """
+        self.machine = machine
+        self.operating_point = operating_point
+        self.solve_Kf = solve_Kf
+        self.solve_Kt = solve_Kt
+        self.solve_Kphi = solve_Kphi
+        self.solve_Kdelta = solve_Kdelta
+        self._validate_attr()
+        
+    def _validate_attr(self):
+        if not isinstance(self.machine,BSPM_Machine):
+            raise TypeError(
+                'Invalid machine type, must be BSPM_Machine.'
+                )
+        
+        if not isinstance(self.operating_point, BSPM_Machine_Oper_Pt):
+            raise TypeError(
+                'Invalid settings type, must be BSPM_Machine_Oper_Pt.'
+                )
+
+class BSPMMachineConstantAnalyzer(BSPM_EM_Analyzer):
+    def __init__(
+            self,
+            configuration: JMAG_2D_Config,
+            Kf_Kt_step: int = 10,
+            Kdelta_coords: list = [[x, y] for x in np.linspace(-0.3,0.3,3) 
+                                   for y in np.linspace(-0.3,0.3,3)],
+            Kphi_step: int = 10,
+            ): 
+        self.configuration = configuration
+        super().__init__(self.configuration)
+        self.Kf_Kt_step = Kf_Kt_step
+        self.Kdelta_coords = Kdelta_coords
+        self.Kphi_step = Kphi_step
```

Comment:
(This is optional) As I understand it, you have prepared 10 "cases" to analyze torque/force at different currents. I would like to suggest using the term 'case' instead of 'step' (generally represents the rotor's angle)
```suggestion
            Kf_Kt_case: int = 10,
            Kdelta_coords: list = [[x, y] for x in np.linspace(-0.3,0.3,3) 
                                   for y in np.linspace(-0.3,0.3,3)],
            Kphi_case: int = 10,
            ): 
        self.configuration = configuration
        super().__init__(self.configuration)
        self.Kf_Kt_case = Kf_Kt_case
        self.Kdelta_coords = Kdelta_coords
        self.Kphi_case = Kphi_case
```